OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -257,22 +257,24 @@ protected void ReserveSpawnBuilding()
 			if (afld == null)
 				return;
 
-			var res = afld.TraitOrDefault<Reservable>();
+			MakeReservation(afld);
+		}
 
-			if (res != null)
-			{
-				UnReserve();
-				Reservation = res.Reserve(afld, self, this);
-			}
+		public void MakeReservation(Actor target)
```

Comment:
More like TryMakeReservation() ?
Because it isn't unlikely for the reservation to fail/it's not guaranteed to happen.